libtransmission/torrent-magnet.cc

```diff
@@ -236,6 +236,42 @@ static void tr_buildMetainfoExceptInfoDict(tr_torrent_metainfo const& tm, tr_var
     }
 }
 
+bool tr_torrentUseMetainfoFromFile(
+    tr_torrent* tor,
+    tr_torrent_metainfo const* metainfo,
+    std::string_view filename_in,
+    tr_error** error)
+{
+    // add .torrent file
+    auto const oldpath = tr_pathbuf{ filename_in };
+    auto const newpath = tr_pathbuf{ tor->torrentFile() };
```

Comment:
No need to create pathbufs for these since both `filename_in` and `tor->torrentFile()` are string_view-friendly:

```suggestion
    auto const oldpath = filename_in;
    auto const newpath = tor->torrentFile();
```